src/Ast/HasType.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+declare(strict_types=1);
+
+namespace Google\Generator\Ast;
+
+class TypeDeclaration {
```

Comment:
This should be in a separate class file, and we can add other constants to it there.

Reply:
Good catch! 

I updated my code to make use of `ResolvedType`. Should we remove this one then? I vote yes.

Reply:
I decided to remove this one in favor for the `ResolvedType` type